plugins/hfp_ag_bluez5.c

```diff
@@ -95,7 +110,10 @@ static DBusMessage *profile_new_connection(DBusConnection *conn,
 		goto invalid;
 
 	dbus_message_iter_get_basic(&entry, &fd);
+
 	dbus_message_iter_next(&entry);
+	if (dbus_message_iter_get_arg_type(&entry) != DBUS_TYPE_ARRAY)
+		goto invalid;
 
 	if (fd < 0)
 		goto invalid;
```

Comment:
Small nit here: these 2 lines should be now before the check for DBUS_TYPE_ARRAY, and you should close fd if the said check fails.